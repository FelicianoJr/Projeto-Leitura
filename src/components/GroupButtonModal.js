import React from "react";

const GroupButtonModal = ({ toggle, pristine, submitting }) => {
  return (
    <div className="modal-footer">
      <button
        type="button"
        onClick={toggle}
        className="btn btn-danger btn-sm"
      >
        Fechar
      </button>
      <button
        type="submit"
        disabled={pristine || submitting}
        className="btn btn-info btn-sm"
      >
        Salvar
      </button>
    </div>
  );
};

export default GroupButtonModal;
