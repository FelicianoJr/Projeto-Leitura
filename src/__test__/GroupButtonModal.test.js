import React from "react";
import { shallow } from "enzyme";
import GroupButtonModal from "../components/GroupButtonModal";

const setup = (pristine, submitting) => {
  const action = {
    toggle: jest.fn()
  };
  const component = shallow(
    <GroupButtonModal {...action} pristine={pristine} submitting={submitting} />
  );

  return {
    component: component,
    action: action,
    button: component.find("button")
  };
};

describe("<GroupButtonModal/>", () => {
  it("should rende  Button", () => {
    const { button } = setup();
    expect(button.length).toEqual(2);
  });

  it("should click Button toggle", () => {
    const { button, action } = setup();
    button.at(0).simulate("click");
    expect(action.toggle).toBeCalled();
  });

  it("should rende disabled when True", () => {
    const { button } = setup(true, false);
    expect(button.at(1).prop("disabled")).toEqual(true);
  });

  it("should rende disabled when false", () => {
    const { button } = setup(false, false);
    expect(button.at(1).prop("disabled")).toEqual(false);
  });

  it("should component Snapshot", () => {
    const { component } = setup();
    expect(component).toMatchSnapshot();
  });
});
