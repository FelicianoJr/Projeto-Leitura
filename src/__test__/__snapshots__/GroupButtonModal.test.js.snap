// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GroupButtonModal/> should component Snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GroupButtonModal
    pristine={undefined}
    submitting={undefined}
    toggle={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="btn btn-danger btn-sm"
          onClick={[Function]}
          type="button"
>
          Fechar
</button>,
        <button
          className="btn btn-info btn-sm"
          disabled={undefined}
          type="submit"
>
          Salvar
</button>,
      ],
      "className": "modal-footer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Fechar",
          "className": "btn btn-danger btn-sm",
          "onClick": [Function],
          "type": "button",
        },
        "ref": null,
        "rendered": "Fechar",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Salvar",
          "className": "btn btn-info btn-sm",
          "disabled": undefined,
          "type": "submit",
        },
        "ref": null,
        "rendered": "Salvar",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="btn btn-danger btn-sm"
            onClick={[Function]}
            type="button"
>
            Fechar
</button>,
          <button
            className="btn btn-info btn-sm"
            disabled={undefined}
            type="submit"
>
            Salvar
</button>,
        ],
        "className": "modal-footer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Fechar",
            "className": "btn btn-danger btn-sm",
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": "Fechar",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Salvar",
            "className": "btn btn-info btn-sm",
            "disabled": undefined,
            "type": "submit",
          },
          "ref": null,
          "rendered": "Salvar",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
